src/services/counters/Counter.ts

```diff
@@ -0,0 +1,63 @@
+import {ICounter} from "./ICounter";
+import {isNotNullOrUndefined} from "../../common/utils/Utils";
+
+export class Counter implements ICounter {
+
+  public static STRONG_TYPE: string = "STRONG";
+  public static WEAK_TYPE: string = "WEAK";
+
+  constructor(private name: string, private type: string, private storage: string,
+              private initialValue: number, private currentValue: number,
+              private lowerBound: number, private upperBound: number,
+              private concurrency: number, private dmr: any) {
+  }
+
+  getStorage(): string {
+    return this.storage;
+  }
+
+  getInitialValue(): number {
+    return this.initialValue;
+  }
+
+  getCurrentValue(): number {
+    return this.currentValue;
+  }
+
+  getLowerBound(): number {
+    return this.lowerBound;
+  }
+
+  getUpperBound(): number {
+    return this.upperBound;
+  }
+
+  getName(): string {
+    return this.name;
+  }
+
+  getConcurrency(): number {
+    return this.concurrency;
+  }
+
+  getType(): string {
+    return this.type;
+  }
+
+  isStrong(): boolean {
+    return this.getType() === Counter.STRONG_TYPE;
+  }
+
+  isWeak(): boolean {
+    return !this.isStrong();
+  }
```

Comment:
This seems like a case for inheritance. `AbstractCounter`, `StrongCounter extends AbstractCounter` etc.

Reply:
Right, good call, let me see how it looks and I'll update the PR